lib/libc/newlib/CMakeLists.txt

```diff
@@ -37,39 +37,43 @@ zephyr_compile_definitions(_ANSI_SOURCE)
 # used by the network stack
 zephyr_compile_definitions(__LINUX_ERRNO_EXTENSIONS__)
 
-# We are using
-# - ${CMAKE_C_LINKER_WRAPPER_FLAG}${CMAKE_LINK_LIBRARY_FLAG}c
-# - ${CMAKE_C_LINKER_WRAPPER_FLAG}${CMAKE_LINK_LIBRARY_FLAG}gcc
-# - c
-# in code below.
-# This is needed because when linking with newlib on aarch64, then libgcc has a
-# link dependency to libc (strchr), but libc also has dependencies to libgcc.
-#
-# CMake is capable of handling circular link dependencies for CMake defined
-# static libraries, which can be further controlled using LINK_INTERFACE_MULTIPLICITY.
-# However, libc and libgcc are not regular CMake libraries, and is seen as linker
-# flags by CMake, and thus symbol de-duplications will be performed.
-# CMake link options cannot be used, as that will place those libs first on the
-# linker invocation. -Wl,--start-group is problematic as the placement of -lc
-# and -lgcc is not guaranteed in case later libraries are also using
-# -lc / -libbgcc as interface linker flags.
-#
-# Thus, we resort to use `${CMAKE_C_LINKER_WRAPPER_FLAG}${CMAKE_LINK_LIBRARY_FLAG}`
-# as this ensures the uniqueness and thus avoids symbol de-duplication which means
-# libc will be followed by libgcc, which is finally followed by libc again.
+if(CONFIG_COMPILER_RT_RTLIB)
```

Comment:
@tejlmand I wonder if `if(${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "llvm")` would be better here 

Similar to:
https://github.com/zephyrproject-rtos/picolibc/blob/1a5c603b9f8e228f9459bdafedad15ea28efc700/zephyr/zephyr.cmake#L99-L103

Reply:
I've changed that